### PR TITLE
COMPASS-4077 - Allow loading Apple notarization credentials from file

### PR DESCRIPTION
**Pull request by addaleax**
This way it’s easier to keep them out of the debug log in CI.

### CODE REVIEW

**Comment by mcasimir** [comment]
> This way it’s easier to keep them out of the debug log in CI.

What is logging that?

**Comment by addaleax** [comment]
@mcasimir A few commands in the compass build process are dumping the environment, you can e.g. grep for `APPLE_PASSWORD` in https://evergreen.mongodb.com/task/10gen_compass_master_macos_oneshot_compile_test_package_publish_patch_d984c87ce8e073ab2bd75e113c85134967577fe4_5f75ee98c9ec441e0a4a1120_20_10_01_14_58_52 (don’t worry, it’s revoked by now :slightly_smiling_face:)

I would say that ideally we should not be logging the environment variables, because it’s always possible that sensitive data ends up in the log, but for now this seems like a good way to solve this specific problem here

File: lib/target.js [hunk 1]
```diff
@@ -529,8 +529,20 @@ class Target {
               if (_err) {
                 return reject(_err);
               }
-              const appleUsername = process.env.APPLE_USERNAME;
-              const applePassword = process.env.APPLE_PASSWORD;
+              let appleUsername = process.env.APPLE_USERNAME;
+              let applePassword = process.env.APPLE_PASSWORD;
+              const appleCredentialsFile = process.env.APPLE_CREDENTIALS;
```
[the remark this answers is not in the document]
Comment (addaleax) [reply lib/target.js]: @anemy sure, done :)

**Comment by addaleax** [comment]
> Maybe a silly question, but where is the file stored? When does the credentials file make it into the evergreen macOS box? 

So, this is my current WIP patch for the main `compass` repo building on top of this PR:

```diff
diff --git a/.evergreen.yml b/.evergreen.yml
index 984fc2535b0d..9f609425ccbf 100644
--- a/.evergreen.yml
+++ b/.evergreen.yml
@@ -259,8 +259,11 @@ functions:
           export NOTARY_AUTH_TOKEN="${signing_auth_token}"
           export NOTARY_SIGNING_KEY="${signing_key_name}"
           export NOTARY_SIGNING_COMMENT="Evergreen project mongodb/compass ${revision} - ${build_variant} - ${branch_name}"
-          export APPLE_USERNAME="${apple_username}"
-          export APPLE_PASSWORD="${apple_password}"
+
+          if [ `uname` == Darwin ]; then
+            echo '{"appleUsername": "${apple_username}", "applePassword": "${apple_password}"}' > ./compass-apple-cred.json
+            export APPLE_CREDENTIALS_FILE="$(pwd)/compass-apple-cred.json"
+          fi
 
           echo "Creating signed release build..."
 
```

It’s not tested yet, but I think it might answer these questions? The main point is just that we don’t want the actual credentials in the environment here, at least for now

**Pull request merged**